core/src/main/scala/com/microsoft/azure/synapse/ml/fabric/OpenAIFabricSetting.scala

```diff
@@ -0,0 +1,48 @@
+package com.microsoft.azure.synapse.ml.fabric
+
+import spray.json.{JsValue, JsString}
+
+trait FabricTenantSetting extends RESTUtils {
+
+  private def getHeaders: Map[String, String] = {
+    Map(
+      "Authorization" -> s"Bearer ${TokenLibrary.getAccessToken}",
+      "Content-Type" -> "application/json"
+    )
+  }
+
+  def usagePost(url: String, body: String): JsValue = {
+    usagePost(url, body, getHeaders);
+  }
+
+  def getModelStatus(modelName: String): Boolean = {
+
+    val payload =
+      s"""["${modelName}"]"""
+
+    val mlWorkloadEndpointML = FabricClient.MLWorkloadEndpointML
+    val url = mlWorkloadEndpointML + "cognitive/openai/tenantsetting"
+    val modelStatus = usagePost(url, payload).asJsObject.fields.get(modelName.toLowerCase).get
+
+    // Allowed, Disallowed, DisallowedForCrossGeo, ModelNotFound, InvalidResult
+    val resultString: String = modelStatus match {
+      case JsString(value) => value
+    }
```

Comment:
this case match will be an error is theres no match, you should handle the _ case here with a RuntimeError